File: go/crypto-square/crypto_square.go
// Crypto Square exercise.
package cryptosquare

import (
	"math"
	"regexp"
	"strings"
)

// Encode given text using the Crypto-Square algorithm.
// @param pt: The text to encode.
// @returns: The encoded text with a space between sections.
func Encode(pt string) string {
	re := regexp.MustCompile("[A-Za-z0-9]+")
	sanitized := strings.ToLower(strings.Join(re.FindAllString(pt, -1), ""))
	width := int(math.Ceil(math.Sqrt(float64(len(sanitized)))))
	height := 0
	for width*height < len(sanitized) {
		height++
	}
	for len(sanitized) < width*height {
		sanitized += " "
	}
	rows := make([]string, width)
	for i := 0; i < width; i++ {
		for j := i; j < len(sanitized); j += width {
			rows[i] = rows[i] + string(sanitized[j])
		}
	}
	return strings.Join(rows, " ")
}

File: php/crypto-square/CryptoSquare.php
<?php
// Crypto Square Exercise.

declare(strict_types=1);

// Encode the given input string using the crypto square algorithm.
// @param $plaintext: The text to encode.
// @returns: The encoded text with a space between each grouping.
function crypto_square(string $plaintext): string
{
    $sanitized = strtolower(join("", preg_split("/[^A-Za-z0-9]+/", $plaintext)));
    $width = intval(ceil(sqrt(strlen($sanitized))));
    $height = 0;
    while ($width * $height < strlen($sanitized)) {
        $height++;
    }
    while (strlen($sanitized) < $width * $height) {
        $sanitized = $sanitized . " ";
    }
    $rows = array();
    for ($j = 0; $j < $width; $j++) {
        $rows[] = "";
        for ($i = $j; $i < strlen($sanitized); $i += $width) {
            $rows[$j] = $rows[$j] . $sanitized[$i];
        }
    }
    return implode(" ", $rows);    
}
